crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2289,24 +2297,33 @@ impl<'db> TypeInferenceBuilder<'db> {
                 .map(Type::IntLiteral)
                 .unwrap_or_else(|| builtins_symbol_ty(self.db, "int").to_instance(self.db)),
 
+            // Division by zero
+            // TODO: handle division by zero for floats
+            (
+                Type::IntLiteral(_),
+                Type::IntLiteral(0),
+                ast::Operator::Div | ast::Operator::FloorDiv | ast::Operator::Mod,
+            ) => {
+                self.division_by_zero_diagnostic(binary.into(), left_ty);
+                Type::Unknown
```

Comment:
Not sure if we should infer `Unknown` here, or go ahead and infer `builtins_symbol_ty(db, "float")` (or int for FloorDiv) despite the error? No other type would be reasonable for checking subsequent code.

Reply:
I'm not sure. I defer to you. It seems fairly reasonable either way, though not using `Unknown` lets us check more code in a single invocation.

Reply:
Yeah, I think I'd favor inferring float or int rather than Unknown here.